src/btree/bt_delete.c

```diff
@@ -293,19 +293,31 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
  *     IDs in the delete info. Since we're called at the end of recovery there's no need to lock the
  *     ref or worry about races.
  */
-static void
+static int
 __delete_redo_window_cleanup_internal(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_REF *child;
 
     WT_ASSERT(session, F_ISSET(ref, WT_REF_FLAG_INTERNAL));
     if (ref->page != NULL) {
         WT_INTL_FOREACH_BEGIN (session, ref->page, child) {
-            if (child->state == WT_REF_DELETED && child->page_del != NULL)
+            /*
+             * We used to do the cleanup when the state of the child was WT_REF_DELETED, but it can
+             * also be loaded in memory if the page has been read before, thus we no longer need
+             * this check anymore.
+             */
+            if (child->page_del != NULL)
                 __cell_redo_page_del_cleanup(session, ref->page->dsk, child->page_del);
+            /*
+             * We still want to make sure reconciliation will remove any stale data, mark it for
+             * eviction.
+             */
+            WT_RET_BUSY_OK(__wt_page_release_evict(session, ref, 0));
```

Comment:
We need to evict the `child` page. Also, when the `child->state` is `WT_REF_MEM`. I believe you cannot call the release_evict directly without taking a hazard pointer. It would help if you had the code something like the one below.
```
    WT_RET(__wt_hazard_set(session, ref, &busy));
    if (busy)
        return (0);
```
Also, we need a retry logic to let the page be evicted as we cannot leave the busy page as it has stale transaction ids. Yield the CPU and check page is still in memory and try again.

Reply:
Thanks for the suggestions, see 500a7a600e6a14d3f372f8150d306ffb5455fbfc